### DISCUSSION TITLE
Removing replication metadata from deleted jobs

### DISCUSSION THREAD

**Discussion by 9ary** [zrepl/zrepl#656] [Q&A] [answered]
I've just replaced my temporary backup pools with a new one, and I would like to clean up the leftover holds and cursors.

There doesn't seem to be a command that does that specifically, and `zrepl zfs-abstraction release-stale` does nothing.

How can I achieve this? Or is it okay to just leave the metadata in place, ie will it just get cleaned up as older snapshots are pruned?

**Comment by 9ary** [comment]
I've figured out how to clean these up manually, they are just bookmarks and holds and can be removed with `zfs destroy` and `zfs release`. The question still stands though: what happens when I remove a job from the config file, and don't clean up? And what is the recommended way to permanently remove a job?

**Comment by problame** [comment, marked as answer]
This will remove all zrepl ZFS abstractions of job `REMOVED_JOB` on all present filesystems:

`zrepl zfs-abstraction release-all --job REMOVED_JOB`

If you want to constraint it to a subset of filesystems, use the `--fs` argument.

Use `zrepl zfs-abstraction release--all --help` to learn more.

**Reply by 9ary** [reply]
Ah, thank you. I missed that there was extra help available.
